### DISCUSSION TITLE
Importing Extracted Data taking too long

### DISCUSSION THREAD

**Discussion by Papricka95** [osm-search/Nominatim#3606] [Q&A]
Hello! im to using nominatim for import osm data. Also i tried to import some files (asia-latest.osm.pbf and europe-latest.osm.pbf) and it was terribly long... I got killed process (status D from htop and low consuming resources of processors <0.2%).

My second try is import first file - asia-latest.osm.pbf. This import was success. But if i try to import europe-latest.osm.pbf as first file i got error and kill this process..

Also i try to "nominatim import --osm-file antarctica-latest.osm.pbf", after i try "nominatim add-data --file ukraine-latest.osm.pbf" i got error and stopping process add-data... And i try "nominatim add-data --file kazakhstan-latest.osm.pbf" and i have same error..

And as result i have so much lost time and i decided to import all planet and this import was success. Why did it happend ? I don't know..
If who know this problem or how it resolve, write to me, pls in this comment

I have configuration PC:
AMD Ryzen 7 5700G, 8 cores, 16 threads
32GB RAM,
SSD disks (2): 2TB, 128 GB for system

**Comment by Papricka95** [comment]
Also when I tried to import the file "europe-latest.osm.pbf" for some reason it couldn't be imported after importing the file "asia-latest.osm.pbf". The first file was imported by the command "nominatim import --osm-file FILE 2>&1 | tee setup.log", and the second file (Europe) was imported by the command "nominatim add-data --file FILE", but after the second command the process froze and for two days did not take up any computer resources, was in the waiting status. I had to kill this process and start importing the entire planet, as I described above.

**Comment by Papricka95** [comment]
I tried to import countries using the same algorithm, since the size is much smaller than the size of the continents, but nevertheless, all processes still hung. It was even impossible to access the disk on which the files were imported via the terminal.

**Comment by mtmail** [comment]
> got error and stopping process 

What was the error message?

Can you check the Postgresql server logfile, maybe it crashed or warned about low resources.

> status D from htop and low consuming resources of processors <0.2%

How much memory did the Nominatim and the database consume? Did the machine swap memory to disk?

> I had to kill this process and start importing the entire planet, as I described above.

I'm not sure the entire planet can be imported with 32GB of memory. For continents (e.g. Europe) and planet make sure you set the flatnode file, see https://nominatim.org/release-docs/latest/admin/Import/#flatnode-files

A common issue is that users copy the postgresql settings from https://nominatim.org/release-docs/latest/admin/Installation/#tuning-the-postgresql-database and don't adapt them to their hardware. The settings on that page are for a 128GB memory machine. Postgresql would try to consume too much on small machines and the operating system would start to swap memory to disk.
